src/core/libraries/dialogs/ime_dialog_ui.cpp

```diff
@@ -0,0 +1,488 @@
+    Free();
+}
+
+ImeDialogState::ImeDialogState(ImeDialogState&& other) noexcept
+    : userId(other.userId), is_multiLine(other.is_multiLine), is_numeric(other.is_numeric),
+      type(other.type), enter_label(other.enter_label), text_filter(other.text_filter),
+      keyboard_filter(other.keyboard_filter), max_text_length(other.max_text_length),
+      text_buffer(other.text_buffer), title(other.title), placeholder(other.placeholder),
+      input_changed(other.input_changed) {
+
+    std::memcpy(current_text, other.current_text, sizeof(current_text));
+
+#ifndef _WIN32
+    orbis_to_utf8 = other.orbis_to_utf8;
+    utf8_to_orbis = other.utf8_to_orbis;
+
+    other.orbis_to_utf8 = (iconv_t)-1;
+    other.utf8_to_orbis = (iconv_t)-1;
+#endif
+
+    other.text_buffer = nullptr;
+    other.title = nullptr;
+    other.placeholder = nullptr;
+}
+
+ImeDialogState& ImeDialogState::operator=(ImeDialogState&& other) {
+    if (this != &other) {
+        Free();
+
+        userId = other.userId;
+        is_multiLine = other.is_multiLine;
+        is_numeric = other.is_numeric;
+        type = other.type;
+        enter_label = other.enter_label;
+        text_filter = other.text_filter;
+        keyboard_filter = other.keyboard_filter;
+        max_text_length = other.max_text_length;
+        text_buffer = other.text_buffer;
+        title = other.title;
+        placeholder = other.placeholder;
+        input_changed = other.input_changed;
+
+        std::memcpy(current_text, other.current_text, sizeof(current_text));
+
+#ifndef _WIN32
+        orbis_to_utf8 = other.orbis_to_utf8;
+        utf8_to_orbis = other.utf8_to_orbis;
+
+        other.orbis_to_utf8 = (iconv_t)-1;
+        other.utf8_to_orbis = (iconv_t)-1;
+#endif
+
+        other.text_buffer = nullptr;
+        other.title = nullptr;
+        other.placeholder = nullptr;
+    }
+
+    return *this;
+}
+
+bool ImeDialogState::CopyTextToOrbisBuffer() {
+    if (!text_buffer) {
+        return false;
+    }
+
+    std::size_t text_len = std::char_traits<char>::length(current_text);
+    return ConvertUTF8ToOrbis(current_text, text_len, text_buffer, max_text_length);
+}
+
+bool ImeDialogState::CallTextFilter() {
+    if (!text_filter || !input_changed) {
+        return true;
+    }
+
+    input_changed = false;
+
+    char16_t src_text[ORBIS_IME_DIALOG_MAX_TEXT_LENGTH + 1] = {0};
+    u32 src_text_length = std::strlen(current_text);
+    char16_t out_text[ORBIS_IME_DIALOG_MAX_TEXT_LENGTH + 1] = {0};
+    u32 out_text_length = ORBIS_IME_DIALOG_MAX_TEXT_LENGTH;
+
+    if (!ConvertUTF8ToOrbis(current_text, src_text_length, src_text,
+                            ORBIS_IME_DIALOG_MAX_TEXT_LENGTH)) {
+        LOG_ERROR(Lib_ImeDialog, "Failed to convert text to orbis encoding");
+        return false;
+    }
+
+    auto* linker = Common::Singleton<Core::Linker>::Instance();
+    int ret =
+        linker->ExecuteGuest(text_filter, out_text, &out_text_length, src_text, src_text_length);
```

Comment:
```suggestion
    int ret = text_filter(out_text, &out_text_length, src_text, src_text_length);
```

Reply:
Why? All guest code is supposed to be called through `ExecuteGuest` now.